### PR TITLE
Implement throttle setting for MAV_CMD_DO_CHANGE_SPEED

### PR DESCRIPTION
**Pull request by sanderux**
Fixes https://github.com/PX4/Firmware/issues/4786

This implements the throttle value of the mission command "Change speed"
Attaching this to the no-airspeed PR as it only has (serious) effect when disabling airspeed sensor

### CODE REVIEW

**Comment by LorenzMeier** [comment]
@tumbili Can you review? Looks good for me.

**Comment by dagar** [comment]
Have you flown this outside @sanderux? I'm just curious how consistently you actually keep the desired throttle? If the commanded throttle isn't sufficient for the altitude setpoint it's going to be ignored anyway.
https://github.com/PX4/Firmware/blob/master/src/lib/external_lgpl/tecs/tecs.cpp#L312

**Comment by sanderux** [comment]
@dagar i havent flown it outside yet. it's true that tecs will balance the throttle with the altitude but there does seem to be a limit to that.

**Comment by LorenzMeier** [comment]
@sanderux Will merge once flown

File: src/modules/navigator/navigator.h [hunk 1]
```diff
@@ -178,6 +178,19 @@ class Navigator : public control::SuperBlock
...
+	 * Get the target throttle
+	 *
+	 * @return the desired throttle for this mission
+	 */
+	float		get_cruising_throttle();
+
+	/**
+	 * Set the target throttle
+	 */
+	void		set_cruising_throttle(float percent=-1.0f) { _mission_throttle = percent; }
```
Comment (tumbili) [inline src/modules/navigator/navigator.h]: @sanderux Why is the argument called 'percent'? Aren't you passing a throttle value?

Comment (sanderux) [inline src/modules/navigator/navigator.h]: @tumbili no, QGC asks for a percentage. mavlink also specs this: https://pixhawk.ethz.ch/mavlink/#MAV_CMD_DO_CHANGE_SPEED

Comment (tumbili) [inline src/modules/navigator/navigator.h]: @sanderux Yeah, the mission item hold a throttle percentage but the function 'set_cruising_throttle' takes as argument a throttle value between 0 and 1.

**Comment by tumbili** [comment]
@LorenzMeier @sanderux @dagar Did a review and it looks good (did a minor comment on variable naming). So next thing here is a log from a flight test.

File: src/modules/navigator/navigator.h (hunk 1 above)
Comment (sanderux) [inline src/modules/navigator/navigator.h]: Ok, shall i just call it throttle, or value then?

Comment (tumbili) [inline src/modules/navigator/navigator.h]: @sanderux throttle is good.

**Comment by tumbili** [comment]
@sanderux Personally I think this change will not change much. TECS will adjust its throttle integrator until it's back at the old desired value. You can verify this in SITL by the way.

**Comment by dagar** [comment]
Right, it's the same as just changing FW_THR_CRUISE.

**Comment by sanderux** [comment]
@tumbili ready to go

**Comment by sanderux** [comment]
@dagar applying this will only make sense when airspeed sensor is disabled. at that point tecs will disregard airspeed and only change throttle if required for altitude. This will enable vtol's with o airspeed sensor to decrease speed before backtransition or increase speed when climbing significantly. a typical mission would be:
- vtol takeoff 20m
- throttle to 80%
- climb to 150m
- throttle to 40%
- cruise 100km
- descend to 20m
- throttle to 20%
- vtol_land

**Comment by tumbili** [comment]
Closing in favour of #4770

**Pull request closed without merging**